x-pack/plugins/monitoring/public/components/elasticsearch/nodes/nodes.js

```diff
@@ -177,6 +179,60 @@ const getColumns = (showCgroupMetricsElasticsearch, setupMode, clusterUuid, aler
     },
   });
 
+  cols.push({
+    name: i18n.translate('xpack.monitoring.elasticsearch.nodes.rolesColumnTitle', {
+      defaultMessage: 'Roles',
+    }),
+    field: 'roles',
+    render: (roles) => {
+      if (!roles) {
+        return i18n.translate('xpack.monitoring.formatNumbers.notAvailableLabel', {
+          defaultMessage: 'N/A',
+        });
+      }
+
+      if (roles.length === 0) {
+        return (
+          <EuiBadge>
+            {i18n.translate('xpack.monitoring.elasticsearch.nodes.coordinatingNodeLabel', {
+              defaultMessage: 'coordinating only',
+            })}
+          </EuiBadge>
+        );
+      }
+
+      if (roles.length > 5) {
+        const head = roles.slice(0, 5);
+        const tail = roles.slice(5, roles.length);
+
+        return (
+          <EuiBadgeGroup gutterSize="xs">
+            {head.map((role) => (
+              <EuiBadge color={role === 'master' ? 'hollow' : 'default'}>{role}</EuiBadge>
+            ))}
+            <EuiToolTip
+              anchorProps={{
+                style: { lineHeight: '1' },
+              }}
+              position="bottom"
+              content={tail.join(', ')}
+            >
+              <EuiBadge>+{tail.length}</EuiBadge>
+            </EuiToolTip>
+          </EuiBadgeGroup>
+        );
+      }
+
+      return (
+        <EuiBadgeGroup gutterSize="xs">
+          {roles.map((role) => (
+            <EuiBadge color={role === 'master' ? 'hollow' : 'default'}>{role}</EuiBadge>
+          ))}
+        </EuiBadgeGroup>
+      );
```

Comment:
**suggestion(non-blocking)**: not sure if you'd like it, but to avoid redeclaring the Badge group you could write this as the default return statement such that:
```js
const head = roles.slice(0, 5);
const tail = roles.slice(5);
const hasMoreRoles = tail.length > 0;

return (
  <EuiBadgeGroup gutterSize='xs'>
    {head.map((role) => (
      <EuiBadge color={role === 'master' ? 'hollow' : 'default'}>{role}</EuiBadge>
    ))}
    {hasMoreRoles && (
      <EuiToolTip
        anchorProps={{
          style: { lineHeight: '1' },
        }}
        position='bottom'
        content={tail.join(', ')}
      >
        <EuiBadge>+{tail.length}</EuiBadge>
      </EuiToolTip>
    )}
  </EuiBadgeGroup>
);
```

Reply:
Nice!